### ISSUE TITLE
How to deal when no response is returned from TM1?

### ISSUE THREAD

**Issue by nicolasbisurgi** [cubewise-code/rushti#66]
Hi,
I'm having a problem when trying to run a taskfile. The processes in the task file should not be retried if failed (mainly because of their incremental nature), as such I put retries=0.

In some occasions, when I execute the taskfile, I get the following error:
```
19376   ERROR   2023-06-23 16:04:35,405   rushti.execute_task  Execution failed. Process: 'lp_extract_avail' with parameters: {'pSeason': 'Current', 'pVersion': 'Refresh', 'pPeriodSub': 'Weeks', 'pPeriod': 'W09', 'pSeasonPosition': 'T'}. Elapsed time: 0:00:02.358255. Error: 'NoneType' object has no attribute 'json'.
```
If this would mean that the process was not called I could add a retry for this case/exception ([as proposed here](https://github.com/cubewise-code/tm1py/pull/766) in the TM1Py repo); but when I check the tm1server.log I see the following:
```
7296   [af]   INFO   2023-06-23 16:04:35.021   TM1.Process   Process "lp_extract_avail" executed by user "LDAP/xxx_tm1_automation ccc_user"
7296   [af]   INFO   2023-06-23 16:04:35.023   TM1.TILogOutput   Process: lp_extract_avail with parameters: {pSeason="Current" pVersion:"Refresh" pPeriodSub="Weeks" pPeriod="W09" pSeasonPosition="T"}
...
7296   [af]   INFO   2023-06-23 16:04:38.006   TM1.Process   Process "lp_extract_avail":  finished executing normally, elapsed time 2.98 seconds
```
Which means that the call was successful and the process execute successfully. So my questions is... how the heck do we account for this? Even if we assume that the process was successful we can't know whether it's done or not (to enable subsequent processes to run).

Do you know why TM1 server does not respond? Is it something that we can improve by tweaking some parameters? Are there any monitoring capabilities that can be implemented to check if the process is still running or if it failed?

Any help is appreciated.

nico

**Comment by nicolasbisurgi** [comment]
**UPDATE**:
The only way to avoid this is to lower down the amount of workers:
With 120 workers: I got ~33 errors out of 119 tasks
With 40 workers: I got ~2 errors out of 119 tasks
With 12 workers: I got 0 errors out of 119 tasks

**Comment by MariusWirtz** [comment]
Hi Nico,

Thanks for raising this issue.
Are you using TM1 in the IBM Cloud?
I wasn't aware that the TM1 operation still happens in this scenario.

I have encountered the same issue using the write_async function with many workers in TM1 in IBM Cloud. I had incorrectly assumed that no response meant that the operation failed.

I think this issue is limited to IBM Cloud and is an issue in the Cloud stack. Something we need to raise with IBM, I think

**Comment by MariusWirtz** [comment]
The issue has been resolved with the recent TM1py 2.0 release   
https://github.com/cubewise-code/tm1py/releases/tag/2.0
